tests/ui/string_slice.rs

```diff
@@ -11,4 +13,6 @@ fn main() {
     let s = String::from(m);
     &s[0..2];
     //~^ ERROR: indexing into a string may panic if the index is within a UTF-8 character
+    let a = Cow::Borrowed("foo");
+    &a[0..3];
```

Comment:
Can you also add an `//~^ ERROR` annotation here (for consistency sake with the other cases in this file)?